src/libexpr/tests/yaml.cc

```diff
@@ -0,0 +1,161 @@
+                for (auto testCase : testCases.listItems()) {
+                    Value *json = nullptr;
+                    bool fail = false;
+                    ctr++;
+                    std::string_view yamlRaw;
+                    for (auto attr = testCase->attrs->begin(); attr != testCase->attrs->end(); attr++) {
+                        auto name = state.symbols[attr->name];
+                        if (name == "json") {
+                            json = attr->value;
+                        } else if (name == "yaml") {
+                            yamlRaw = state.forceStringNoCtx(*attr->value, noPos, "while interpreting the \"yaml\" field as string");
+                        } else if (name == "fail") {
+                            fail = state.forceBool(*attr->value, noPos, "while interpreting the \"fail\" field as bool");
+                        }
+                    }
+                    fail |= !json;
+                    bool emptyJSON = false;
+                    std::string_view jsonStr;
+                    Value jsonVal;
+                    std::string yamlStr = replaceUnicodePlaceholders(yamlRaw);
+                    Value yaml, yamlVal;
+                    Value *pYaml = &yaml;
+                    yaml.mkString(yamlStr);
+                    if (!fail) {
+                        if (json->type() == nNull) {
+                            jsonStr = "null";
+                            jsonVal.mkNull();
+                        } else {
+                            jsonStr = state.forceStringNoCtx(*json, noPos, "while interpreting the \"json\" field as string");
+                        }
+                        if (!(emptyJSON = jsonStr.empty())) {
+                            if (json->type() != nNull) {
+                                jsonVal = parseJSONStream(state, jsonStr, fromYAML);
+                                jsonStr = printValue(state, jsonVal);
+                            }
+                            fromYAML(state, noPos, &pYaml, yamlVal);
+                            EXPECT_EQ(printValue(state, yamlVal), printValue(state, jsonVal)) << "Testcase[" + std::to_string(ctr) + "]: Parsed YAML does not match expected JSON result";
+                        }
+                    }
+                    if (fail || emptyJSON) {
+                        try {
+                            fromYAML(state, noPos, &pYaml, yamlVal); // should throw exception, if fail is asserted, and has to throw, if emptyJSON is asserted
+                        } catch (const EvalError &e) {
+                            continue;
+                        }
+                    }
+                    EXPECT_FALSE(emptyJSON) << "Testcase[" + std::to_string(ctr) + "]: Parsing empty YAML has to fail";
+                    // ryml parses some invalid YAML successfully
+                    // EXPECT_FALSE(fail) << "Testcase[" << ctr << "]: Invalid YAML was parsed successfully";
+                }
+                return "OK";
```

Comment:
Isn't it cleaner to return a boolean? Then the call sites can just be `ASSERT(execYAMLTest(...))`.